orchestrator/src/fund.go

```diff
@@ -140,4 +142,34 @@ func (FundAction) RunMany(config Config, actionIOs []ActionIO) error {
 	return nil
 }
 
+func (FundAction) Validate(rawParams json.RawMessage) error {
+	var params FundParams
+	if err := json.Unmarshal(rawParams, &params); err != nil {
+		return fmt.Errorf("failed to unmarshal fund-keys params: %v", err)
+	}
+	fundKeysBaseDir := extractBaseDir(params.Prefix)
+	if folderExists(fundKeysBaseDir) {
+		return fmt.Errorf("directory '%s' already exists. Please re-generate script using unique experiment name or different '-fund-keys-dir' CLI argument value", fundKeysBaseDir)
+	}
+	return nil
+}
+
+// Helper function to extract the first two levels of the path
+func extractBaseDir(prefix string) string {
+	parts := strings.Split(filepath.ToSlash(prefix), "/")
+	if len(parts) >= 3 {
+		return strings.Join(parts[:3], "/")
+	}
+	return prefix
+}
+
+// Helper function to check if a folder exists
+func folderExists(path string) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return info.IsDir()
```

Comment:
Maybe we simply run the file exists check?

If the path contains a file (and not a folder), this function would return true, however this is clearly a problem for the funding to continue

Reply:
@georgeee 

> If the path contains a file (and not a folder), this function would return true

I don't think it is correct. If path contains file the function returns `false`.
Or what am I missing here?

Reply:
Yep, sorry, the function will return `false`, and validation check will pass. My point is that if the path is a file, the experiment would fail later (because we won't be able to create a subdir for that file), hence it makes more sense to fail during validation

Reply:
Ah, got it, thank you